edgehog-device-runtime-containers/src/service/mod.rs

```diff
@@ -533,7 +485,35 @@ impl AsRef<str> for Id {
 
 impl Display for Id {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.0)
+        write!(f, "{} ({})", self.rt, self.id)
+    }
+}
+
+/// Type of the container resource [`Id`]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
+#[repr(u8)]
+pub enum ResourceType {
+    /// Image resource.
+    Image = 0,
+    /// Volume resource.
+    Volume = 1,
+    /// Network resource.
+    Network = 2,
+    /// Container resource.
+    Container = 3,
+    /// Deployment resource.
+    Deployment = 4,
+}
```

Comment:
Why are you making explicit the numbers associated to each enum value?

Reply:
To be stable and serializable as numbers